sdk/resourcemanager/docs/THREAD_POOL_AND_CONNECTION_POOL.md

```diff
@@ -0,0 +1,90 @@
+```java readme-sample-azureClientThreadPool
+// Thread pool configuration.
+// Official Reactor Netty documentation for defaults: https://projectreactor.io/docs/netty/release/reference/#client-tcp-level-configurations-event-loop-group
+singletonHttpClientBuilder
+    .eventLoopGroup(LoopResources
+        .create(
+            "client-thread-pool", // thread pool name
+            Runtime.getRuntime().availableProcessors() * 2, // thread pool size
+            true)
+        // we use our custom event loop here, disable the native one
+        .onClient(false))
+    .build();
+```
+
+### <a name="azureidentity"></a>Configure Azure Identity client to use the singleton HttpClient (DefaultAzureCredential for example):
+By default, Azure Identity uses `ForkJoinPool.commonPool()` for token acquisition. The pool size equals the number of processors available to the runtime on initialization minus 1 (with a minimum of 1). It's a singleton pool and is shared by default. Default pool configuration should be sufficient for most use cases.
+```java readme-sample-azureIdentityThreadpool
+// Use the singleton httpClient for Azure Identity
+final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+final TokenCredential credential = new DefaultAzureCredentialBuilder()
+    .authorityHost(profile.getEnvironment().getActiveDirectoryEndpoint())
+    .executorService(ForkJoinPool.commonPool()) // thread pool for executing token acquisition, usually we leave it default
+    .httpClient(singletonHttpClient)
+    .build();
+```
+
+### <a name="azureclient"></a>Configure Azure Clients to use the singleton HttpClient and Azure Identity client:
+```java readme-sample-azureClientHttpClient
+// Use the singleton httpClient for your Azure client
+AzureResourceManager azureResourceManager = AzureResourceManager
+    .configure()
+    .withLogLevel(HttpLogDetailLevel.BASIC)
+    .withHttpClient(singletonHttpClient)
+    .authenticate(credential, profile)
+    .withSubscription(subscriptionId); // your subscription ID, can be different for different Azure clients
+```
+
+### <a name="reactor"></a>Configure Reactor thread pool size:
+By default, Azure Client subscribes on Schedulers.parallel(), which has a thread pool of size equal to available processor count. It is a static pool and is shared in nature.
+You can change the pool size by specifying the variable `reactor.schedulers.defaultPoolSize` in environment variables, whether through command line arguments, or System.setProperty.
+
+
+## Other JVM thread configurations you might be interested:
+[Compiler threads for JIT compiler](https://docs.oracle.com/en/java/javase/17/docs/specs/man/java.html#advanced-jit-compiler-options-for-java)
+[GC threads](https://docs.oracle.com/en/java/javase/17/docs/specs/man/java.html#advanced-garbage-collection-options-for-java)
```

Comment:
md formatting is wrong. Either use a list, or use line break.

Reply:
Got it, will use list.